src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -297,6 +297,10 @@ private async Task<IDictionary<string, string>> ValidateFunctionAppPublish(Site
                             ColoredConsole.WriteLine(WarningColor($"Setting '{Constants.FunctionsWorkerRuntime}' to 'dotnet-isolated'"));
                             result[Constants.FunctionsWorkerRuntime] = WorkerRuntimeLanguageHelper.GetRuntimeMoniker(workerRuntime);
                         }
+                        else if (workerRuntime == WorkerRuntime.None)
+                        {
+                            throw new ArgumentException($"Your local project is set to '{workerRuntime}' is not a valid option. Options are {WorkerRuntimeLanguageHelper.GetAvailableOptions()}");
```

Comment:
Should this be a `CliException` ?

Reply:
@liliankasem , I have updated it to CliException.